hl7_mock_receiver/hl7_mock_receiver/hl7_mock_receiver_application.py

```diff
@@ -4,19 +4,44 @@
 import threading
 from typing import Any
 
-from hl7apy.mllp import MLLPServer
+from hl7apy.core import Message
+from hl7apy.mllp import MLLPRequestHandler, MLLPServer
 from message_bus_lib.connection_config import ConnectionConfig
 from message_bus_lib.servicebus_client_factory import ServiceBusClientFactory
 
 from .app_config import AppConfig
-from .error_handler import ErrorHandler
 from .generic_handler import GenericHandler
 
 log_level_str = os.environ.get("LOG_LEVEL", "INFO").upper()
 log_level = getattr(logging, log_level_str, logging.INFO)
 logging.basicConfig(level=log_level, format="%(asctime)s [%(levelname)s] %(message)s")
 logger = logging.getLogger(__name__)
 
+
+class CustomMLLPRequestHandler(MLLPRequestHandler):
+
+    def _route_message(self, msg: Message) -> Any:
+
+        try:
+            generic_handler_config = self.handlers.get('generic')
+            if not generic_handler_config:
+                raise ValueError("Generic handler configuration not found in handlers")
+
+            handler_class, *handler_args = generic_handler_config
+
+            handler = self._create_handler(handler_class, msg, handler_args)
+
+            return handler.reply()
+        except Exception as e:
+            logger.error("Error routing message: %s", e)
+            try:
+                err_handler, args = self.handlers['ERR'][0], self.handlers['ERR'][1:]
+                h = self._create_error_handler(err_handler, e, msg, args)
+                return h.reply()
+            except KeyError:
+                raise e
```

Comment:
Previously the override was on the `init` function, now it's entirely on _route_message, bit more context would be appreciated as I'm not sure what the intention of this is or what it is trying to fix.
What about the handler in the exception scenario, would it not try to call the ERR one which has been removed from the list of handlers and just raise a KeyError?